web/src/components/tasks/NaicsCodeTask.test.tsx

```diff
@@ -417,14 +417,53 @@ describe("<NaicsCodeTask />", () => {
       expect(screen.getByText(`Register & ${Config.determineNaicsCode.saveButtonText}`)).toBeInTheDocument();
     });
 
-    it("opens Needs Account modal on save button click", async () => {
```

Comment:
This test is still necessary, because I could comment out your code on lines 76-79 and all the tests would still pass. 

However, just bringing back this test as it was written wouldn't be enough because it fires a change event and THAT triggers the modal, so the test-as-written would still pass even when I remove lines 76-79. We need a test that drives out and necessitates that guard clause.

Please add the following test:

```
    it("opens Needs Account modal on save button click", () => {
      renderPage();
      fireEvent.click(screen.getByText(`Register & ${Config.determineNaicsCode.saveButtonText}`));
      expect(setShowNeedsAccountModal).toHaveBeenCalledWith(true);
    });
```